test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperScopeManagerTest.cs

```diff
@@ -17,6 +17,127 @@ public class TagHelperScopeManagerTest
             async () => await Task.FromResult(result: true);
 
         [Fact]
+        public void Begin_DoesNotRequireParentExecutionContext()
```

Comment:
good coverage but worth testing to confirm "does not affect" is shallow.  that is, test parent adding a test class instance to the parent's `Items`, make a change to a property of that instance in child execution context, and confirm parent sees modification.